packages/prettier-plugin-java/src/printers/blocks-and-statements.ts

```diff
@@ -192,12 +193,19 @@ export class BlocksAndStatementPrettierVisitor extends BaseCstPrettierPrinter {
   }
 
   ifStatement(ctx: IfStatementCtx) {
+    handleCommentsIfStatement(ctx);
```

Comment:
Is this issue only happening on if statements ?

Reply:
Looks like it also happens with for loops, while loops, and switch statements. I'll see about fixing those cases as well.

Reply:
Latest push now handles for/while/do loops and switch/catch statements as well.